<launch>
  <!-- PASSED PARAMS-->
  <arg name="debug"
    default="true"  />
  <arg name="load_config"
    default="true"  />
  <arg name="config_file"
    default="default" />
  <arg name="config_path"
    default="$(find parking_spot_detector)/config/$(arg config_file).yaml"  />
  <arg name="node_respawn"
    default="true"  />
  <arg name="node_output"
    default="screen" doc="screen or log" />
  
  <!-- PARAMS FROM YAML -->
  <group if="$(arg load_config)">
    <rosparam command="load"
      file="$(arg config_path)"  />
  </group>


  <node 
    name="parking_spot_detector" 
    type="detect_parking_spot" 
    pkg="parking_spot_detector" 
    respawn="$(arg node_respawn)"
    output="$(arg node_output)" >

    <param name="visualization_in_searching" value="$(arg debug)" />
  </node>

</launch>
